stories/2022-03-27_Glass_ML/ML_20th_Century.ipynb

```diff
@@ -0,0 +1,2194 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    import pandas as pd
</pre><p>Jennifer Ding:</p><ul><li>Also if you are importing libraries pandas the notebook, would be better to just import once and standardise the method (e.g. use import pandas as pd in cell 4 rather than importing just read_csv and then importing pandas as a library in cell 8 and in cell 12, etc. Same with matplotlib).</li></ul><p>Joe Cerniglia:</p><p>There’s definitely some cleanup that I can do here, and I would heartily agree that most often having imports at the top of the code is the best policy, but I would present a counter-argument in certain isolated and limited cases, such as these, that having relevant imports close to the place in which they are used may help those who want to copy and paste code snippets into their own code.&nbsp;The custom diverging colormaps are one example of code that might be useful for readers to try out on their own. Place the imports at the top, many lines distant from their use, and one must search for the correct set of imports later to copy and paste when that snippet is first executed. Also, these imports are only used in the section in which they are located.</p><p> </p><p>See <a href="https://allanderek.github.io/posts/import-placement/" target="_blank">https://allanderek.github.io/posts/import-placement/</a>, in the section titled ‘Bunching Imports with Relevant Code’.</p><p> </p><p>PEP8 states: “However, know when to be inconsistent – sometimes style guide recommendations just aren’t applicable. When in doubt, use your best judgment.”&nbsp;</p><p> </p><p>I will go through the notebook and remove redundant imports and reflect further on where this aspect of the code can be tightened.</p><p> </p><p>However we decide, we may choose to have a comment in the story about why these decisions of placement of import statements were made.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/TuringDataStories/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>